### PR TITLE
Migrate to the C++ version of Yoga

### PR DESCRIPTION
**Pull request by bschwind**
@victorporof @mamaar

I tried to combine changes from both of your PRs to get a minimal version which builds with the new C++ files. I had to remove `-Werror` because it [doesn't build correctly](https://travis-ci.org/bschwind/yoga-rs/jobs/393106699) on Travis. I suspect because it's using g++? Maybe the official Yoga library only uses Clang 🤷‍♂️

@victorporof I also wasn't sure about `.enable_cxx_namespaces()`, I couldn't get it building with that enabled. Could you also check that this works with your emscripten+asm stuff? I don't have a dev environment/workflow set up for those tools.

I want to get just these changes in as a base, and then @victorporof can merge in the other changes related to removing `mem::transmute` and @mamaar can fill in some test automation changes if desired, because we'll need to bring over the new tests from yoga.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -7,23 +7,38 @@ use std::env;
...
 	let bindings = bindgen::Builder::default()
 		.rust_target(RustTarget::Stable_1_21)
-		.blacklist_type("max_align_t") // This fails `cargo test` so disable for now
-		.blacklist_type("FP_INFINITE")
-		.blacklist_type("FP_NAN")
-		.blacklist_type("FP_NORMAL")
-		.blacklist_type("FP_SUBNORMAL")
-		.blacklist_type("FP_ZERO")
+		.no_convert_floats()
+		// .enable_cxx_namespaces()
```
Comment (bschwind) [inline build.rs]: @victorporof I couldn't get the code building with this. I'm familiar with C++ namespaces but I'm not sure what it means to "enable" or "disable" them.

**Comment by bschwind** [comment]
Also if either of you want your name on the commit we can restructure one of your PRs to match the changes here, since I basically pulled from both of them.

**Comment by victorporof** [comment]
@bschwind Don't really care about having my name on this commit.

IIRC `enable_cxx_namespaces` was required to get things working in emscripten, but I'm not sure anymore. Turning that on simply required importing from a bindgen-generated `root` module inside your declared `internal`: `pub use self::root::*;`.

**Comment by bschwind** [comment]
Sounds good, do you think this is good to merge?

**Comment by mamaar** [comment]
I'm good with not having my name on it. Will look at the test translations this week.

**Comment by bschwind** [comment]
Alright, thanks for the help from both of you!

**Pull request merged**